### ISSUE TITLE
Investigation: Is there a way to get more useful client errors from react

### ISSUE THREAD

**Issue by Cooryd** [elifesciences/elife-xpub#2237]
## Background

This investigation stems from issue #2212 where a user provided hotjar feedback which reports the throwing of a minified react error:

```
Minified React error #185; visit https://reactjs.org/docs/error-decoder.html?invariant=185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
```

This isn't very useful information when trying to diagnose client side production issues and on a quick search it seems the only way to enable full react error output is to build in `develop` mode which isn't something we would ever want to deploy. 

We should investigate if there is a better way for us to record errors so we can better identify issues that happen in production that are not easy to reproduce locally.

**Comment by erezmus** [comment]
So far, trying to get source maps for js to output in production mode

**Comment by erezmus** [comment]
Managed to get source maps working after upgrading uglifyjs-webpack-plugin to v2.1.3

The plan is now to:
- replace the (probably outdated) newrelic.js browser agent in app/assets with either a script injected via the node APM or the official npm package
- use the newrelic api to upload the source maps on deploy

**Comment by hdrury1** [comment]
@erezmus AND @Cooryd worth raising these two tasks as separate tickets and closing this investigatory one off?

**Comment by erezmus** [comment]
@hdrury1 there is still something about how newrelic is hooked up that i need to go over, but yes in principle we should do that.

**Comment by erezmus** [comment]
I don't think we have any monitoring on the node.js side with new relic unless i've missed something

**Comment by erezmus** [comment]
Ok so after speaking to @giorgiosironi, the root level app.js file (which includes the newrelic) is actually used but only in production.

https://github.com/elifesciences/elife-xpub-deployment/blob/b4c288219fd82c5c899813e18f1c22df5e3e668d/docker-compose.yml#L9

This runs the pubsweet server only. Hence we have DO have newrelic working but only in production. (ignore previous comment).

**Comment by erezmus** [comment]
Issue created for fix: #2254
